xbmc/rendering/dx/DirectXHelper.cpp

```diff
@@ -196,4 +197,19 @@ const std::string D3D11VideoProcessorFormatSupportToString(
       return std::to_string(value);
   }
 }
+
+AVPixelFormat DxgiToAvPixelFormat(DXGI_FORMAT dxgiFormat)
```

Comment:
I vote for move to static method in `CRendererSoftware` or even as only has two possibilities calculate in place like:

```c++
  const AVPixelFormat dstFormat =
      (target.GetFormat() == DXGI_FORMAT_R10G10B10A2_UNORM) ? AV_PIX_FMT_X2BGR10 : AV_PIX_FMT_BGRA;
```

Also thinking in more compact changes for easy backport

Reply:
Sure, I could split the transformation in 2 steps and backport only the first (using the snippet you suggested)
We have those sort of conversions in multiple places already, I wanted to start centralizing them to avoid more duplication / reeinventing the wheel.

Reply:
on the other hand I don't know at this time if such conversion will be needed somewhere else.

Reply:
> on the other hand I don't know at this time if such conversion will be needed somewhere else.

This is the reason of prefer in place variable only, if "at this time" only is used here always is a more optimized and efficient code. When (in the future) this is used in 2 or more places then it makes sense to create a method but not now.

I think the third commit can be removed for now and added in another PR when are needed. This will make everything easier and clearer.

Reply:
sure no big deal